api/v1alpha1/helpers.go

```diff
@@ -0,0 +1,20 @@
+package v1alpha1
+
+import "k8s.io/apimachinery/pkg/util/yaml"
+
+// *****
+// Helper types
+// *****
+// Channel - list all channels for a given operator
+type Channel struct {
+	Name       string `json:"name"`
+	CurrentCSV string `json:"currentCSV"`
+}
+
+// *****
+// Helper funcs/methods
+// *****
+// FromYAML - instantiates an AddonMetadata struct from yaml data
+func (a *AddonMetadata) FromYAML(data []byte) error {
+	return yaml.Unmarshal(data, a)
+}
```

Comment:
I often used this pattern of `FromXXX` as a method receiver on a type (e.g.: `FromYAML`, `FromJSON`, etc.). Because it is bound to a specific version `v1alpha` I don't see where else I could expose these. As declaring method receivers outside of a package is prohibited... :thinking:

Reply:
Hmm, how about having a helper function like `GetAddonMetadataFromYAML(a *apis.AddonMetadata, data [byte]) error { ... }` ?

Reply:
If we bum to `v1beta1` wouldn't that break this function? seems it should live alongside each CRD version?

Reply:
Yeah, sounds right. Let's just keep this one for now :)